### PR TITLE
docs: improve list of rules, add data to rules metadata

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR improves the page of lint rules. The list of rules is still divided by group, but they are now displayed inside a table and with an emoji that shows if a rule si recommended, if it has a safe fix and if it has an unsafe fix.

To show the entity of the fix, I created a new metadata that stores that information. This information makes `applicability` irrelevant, although that would require another refactor that we can push in another PR.

Also, I updated the diagnostics. to show `Unsafe fix` instead of `Suggested fix`.

## Test Plan

Check the deployed website to make sure that the correct information is in sync with `Applicability` of the rule.
The current CI should still work.

### CODE REVIEW

File: crates/biome_js_analyze/src/analyzers/correctness/no_unused_labels.rs [hunk 1]
```diff
@@ -57,6 +57,7 @@ declare_rule! {
         version: "1.0.0",
         name: "noUnusedLabels",
         recommended: true,
+        fix_kind: FixKind::Unsafe,
```
Comment (victor-teles) [inline crates/biome_js_analyze/src/analyzers/correctness/no_unused_labels.rs:60]: Should be Safe instead as Applicability is `Always`?

File: crates/biome_js_analyze/src/analyzers/style/use_literal_enum_members.rs [hunk 2]
```diff
@@ -67,6 +67,7 @@ declare_rule! {
         version: "1.0.0",
         name: "useLiteralEnumMembers",
         recommended: true,
+        fix_kind: FixKind::Unsafe,
```
Comment (victor-teles) [inline crates/biome_js_analyze/src/analyzers/style/use_literal_enum_members.rs:70]: This one doesn't have a `action`, fix_kind need to be set either ?

File: crates/biome_js_analyze/src/semantic_analyzers/style/no_shouty_constants.rs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 use crate::{semantic_services::Semantic, utils::batch::JsBatchMutation, JsRuleAction};
-use biome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic,
+};
```
Comment (victor-teles) [inline crates/biome_js_analyze/src/semantic_analyzers/style/no_shouty_constants.rs:4]: `fix_kind` is missing in this one

File: crates/biome_js_analyze/src/syntax/nursery/no_super_without_extends.rs [hunk 4]
```diff
@@ -21,6 +21,7 @@ declare_rule! {
         version: "1.0.0",
         name: "noSuperWithoutExtends",
         recommended: false,
+        fix_kind: FixKind::Unsafe,
```
Comment (victor-teles) [inline crates/biome_js_analyze/src/syntax/nursery/no_super_without_extends.rs:24]: This one doesn't have a action, fix_kind need to be set either ?

**Comment by Conaclos** [comment]
Great suggestion!

I think we could use the same emoji that other tools:

- ✅ recommended rule
- 🔧 safe fixes
- 💡 unsafe fixes (suggestions)

**Comment by ematipico** [comment]
> Also, should we add the action category in the metada (quickfix, ...)?

The action category is only used in the LSP at the moment, and we always use quickfix, so I don't think we should move it

**Comment by ematipico** [comment]
> Great suggestion!
> 
> I think we could use the same emoji that other tools:
> 
> - ✅ recommended rule
> - 🔧 safe fixes
> - 💡 unsafe fixes (suggestions)


I thought about it, although I have some reservations about the light bulb. Contrary to Eslint, in Biome users can see unsafe fixes in the diagnostics, and apply them either via CLI or via LSP. Hence, the warning sign

**Comment by ematipico** [comment]
@victor-teles thank you for the double check!

@Conaclos one more thing. As you know, in the documentation, we use the first paragraph as a short description of the rule. Now the paragraph can't be broke on multiple lines anymore because that will cause rendering issues in the table, that's why the PR changes some of them.

**Review by victor-teles — APPROVED** [review]

**Pull request merged**